tools/toolchains/gcc.py

```diff
@@ -238,6 +247,8 @@ def link(self, output, objects, libraries, lib_dirs, mem_map):
         # Exec command
         self.cc_verbose("Link: %s" % ' '.join(cmd))
         self.default_cmd(cmd)
+        if self.target.core == "Cortex-M23" or self.target.core == "Cortex-M33":
+            print "Secure File: %s" %secure_file
```

Comment:
Please use
```python
self.progress("Secure Library Object", secure_file)
```
instead.

Reply:
@theotherjimmy - Build for non-secure will fail if secure library file is missing during linking. To make it more user friendly:
1. Should be scan and report for missing file in advance? 
2. Add it  as requirement in app,json and expect it to be at same location?

Reply:
Hmmmm... We could do that, but what if someone compiles for a pure non-secure CORTEX-M23-NS? Would you be locking them into having an empty object file to trick the build system?

Reply:
Empty object file won;t help because it will again fail during linking

Reply:
You misunderstood my question. Let me rephrase. Can a user build a non-secure Cortex-M23-NS application without any secure component?

Reply:
Core will boot in secure mode always and to switch to non-secure mode, stubs function and switch to non-secure reset handler will be required.  One cannot boot with just non-secure project.

Reply:
Okay. If I understand correctly, A user would need to have an object file compiled for secure Cortex-M23 containing these functions, even if they don't care about security and just want to compile there entire application for Cortex-M23-NS.

If that's the case, then I think it's fine to insist that they have an object file matching the naming convention created by the linker.

Reply:
@theotherjimmy 
> self.progress("Secure Library Object", secure_file)

Does truncates file path ` basename(event['file'])` and prints just filename. I need path of file to be printed. Didn't saw any args which can be helpful.

Reply:
> Does truncates file path basename(event['file']) and prints just filename.

I'm having trouble understanding that this sentence is supposed to mean. Could you rephrase it? Is this a question? Is this an assertion?

> I need path of file to be printed.

Hmmmmmm.... That's not supported by it. Try:
```python
self.info("Secure Library object file %s" % secure_file)
```